File: practice23/14.cpp

//https://www.nowcoder.com/practice/0cff324157a24a7a8de3da7934458e34?tpId=98&tqId=32912&tPage=5&rp=5&ru=/ta/2019test&qru=/ta/2019test/question-ranking
#include<iostream>
using namespace std;

struct Node
{
	int element;
	Node *next;
};

Node* createNode()
{
	Node *pt = new Node;
	Node *current = pt;
	for (int i = 1; i <= 7; i++)
	{
		current->element = i;
		current->next = new Node;
		current = current->next;
	}
	current->next = NULL;
	return pt;
}

void find(int Index, Node *pt)
{
	for (int i = 0; i < 7 - Index; i++)
	{
		pt = pt->next;
	}
	cout << pt->element << endl;
}

int main()
{
	Node *first;
	int n;
	while (cin >> n)
	{
		first = createNode();
		find(n, first);
	}
	system("pause");
	return 0;
}

File: practice30/14.java


//https://www.nowcoder.com/practice/abf0f0d6b4c44676b44e66060286c45a?tpId=90&tqId=30835&tPage=3&rp=3&ru=/ta/2018test&qru=/ta/2018test/question-ranking
import java.util.*;

public class Main {
    public static void main(String args[]) {
        Scanner scanner = new Scanner(System.in);
        String s = scanner.next();
        int x, y;
        int index=s.length();
        String sub;
        for (int i = s.length() - 1; i >= 1; i--) {
            sub=s.substring(i,s.length());
            if(s.startsWith(sub)){
                index=i;
            }
        }
        String str=s.substring(s.length()-index,s.length());
        System.out.println(s+str);
    }
}